src/rust/aggregator/src/data/user_history.rs

```diff
@@ -348,70 +357,94 @@ impl Data for UserHistory {
         }
         if !cancel_events.is_empty() {
             for x in cancel_events {
-                sqlx::query!(
-                    r#"
-                        INSERT INTO aggregator.aggregated_events VALUES (
-                            $1, $2
-                        );
-                    "#,
-                    x.txn_version,
-                    x.event_idx,
-                )
-                .execute(&mut transaction as &mut PgConnection)
-                .await
-                .map_err(|e| DataAggregationError::ProcessingError(anyhow!(e)))?;
-            }
-        }
-        for x in &fill_events {
-            // If remaining_size is 0, update order_status.
-            // But if order_status was 'cancelled', do not update.
-            sqlx::query!(
-                r#"
-                    UPDATE aggregator.user_history
-                    SET
-                        remaining_size = remaining_size - $1,
-                        total_filled = total_filled + $1,
-                        order_status = CASE remaining_size
-                            WHEN 0 THEN CASE order_status
-                                WHEN 'cancelled' THEN order_status
-                                ELSE 'closed'
-                            END
-                            ELSE order_status
-                        END,
-                        last_updated_at = CASE last_updated_at > $4
-                            WHEN true THEN last_updated_at
-                            ELSE $4
-                        END
-                    WHERE order_id = $2 AND market_id = $3;
-                "#,
-                x.size,
-                x.maker_order_id,
-                x.market_id,
-                x.time,
-            )
-            .execute(&mut transaction as &mut PgConnection)
-            .await
-            .map_err(|e| DataAggregationError::ProcessingError(anyhow!(e)))?;
-        }
-        if !fill_events.is_empty() {
-            for x in fill_events {
-                sqlx::query!(
-                    r#"
-                        INSERT INTO aggregator.aggregated_events VALUES (
-                            $1, $2
-                        );
-                    "#,
-                    x.txn_version,
-                    x.event_idx,
-                )
-                .execute(&mut transaction as &mut PgConnection)
-                .await
-                .map_err(|e| DataAggregationError::ProcessingError(anyhow!(e)))?;
+                mark_as_aggregated(&mut transaction, &x.txn_version, &x.event_idx).await?;
```

Comment:
It doesn't seem like we actually do anything with the cancel events other than marking them as aggregated. Shouldn't we be updating the status of the order to `cancelled`?

Reply:
That is done in the loop above. Actually, let me merge the loops as it is useless and confusing that there are two of them.